bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SmartGcal.scala

```diff
@@ -0,0 +1,22 @@
+package edu.gemini.seqexec.odb
+
+import edu.gemini.spModel.core.Peer
+import edu.gemini.spModel.gemini.calunit.smartgcal.{CalibrationProvider, CalibrationRepository}
+import edu.gemini.spModel.smartgcal.UpdatableCalibrationRepository
+import edu.gemini.spModel.smartgcal.provider.CalibrationProviderImpl
+import edu.gemini.spModel.smartgcal.repository.{CalibrationUpdater, CalibrationFileCache, CalibrationRemoteRepository}
+
+import java.nio.file.Path
+import java.time.Duration
+
+object SmartGcal {
+  def initialize(peer: Peer, dir: Path): TrySeq[CalibrationProvider] =
+    catchingAll {
+      val service    = new CalibrationRemoteRepository(peer.host, peer.port)
+      val cachedRepo = new CalibrationFileCache(dir.toFile)
+      val provider   = new CalibrationProviderImpl(cachedRepo)
+      CalibrationUpdater.instance.addListener(provider)
+      CalibrationUpdater.instance.start(cachedRepo, service, Duration.ofHours(1).toMillis)
```

Comment:
Could there be troubles if we do the update at the same exact time that we need to unloop a sequence?